### ISSUE TITLE
🐛 Parsing fails on `CDATA`

### ISSUE THREAD

**Issue by Zearin** [biomejs/biome#11366]
### Environment information

<details>

```bash

CLI:
  Version:                      2.5.8
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_DISTRIBUTION:                npm
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v26.7.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully.
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0

```

</details>

## Bug

I’m working on an SVG file. I get the following error diagnostic from Biome: `Expected a qualified rule, or an at rule but instead found '<!'.`

## Context

In the `style` element, I open a `CDATA` section to satisfy another (XML-based) tool that examines my code.

Since `CDATA` (which stands for “Character Data”) isn’t as common as it used to be, I was worried Biome wouldn’t support it.  Fortunately, I found evidence of it in the Biome codebase:

- **Search results in 56 files (as of 2026-08-16):** https://github.com/search?q=repo%3Abiomejs%2Fbiome+CDATA&type=code
- **Specific instance of CDATA `kind`:** https://github.com/biomejs/biome/blob/6cd32636f463551b8dfe46aeefb1191d437b4694/crates/biome_html_syntax/src/generated/kind.rs#L21

After a moment, I realized I was silly to worry! Browsers still have to respect CDATA when they see it.

### Expected result

There’s already code to _support_ parsing `CDATA`; it’s just not being used.  

The expected result is for the parsing to happen without throwing an error because of this.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
What's the other tool you're using?

**Comment by biome-cookie** [comment]
Hello @Zearin, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by Zearin** [comment]
[Here is a demo on the playground](https://biomejs.dev/playground/?indentWidth=4&experimentalFullSupportEnabled=false&tailwindDirectives=false&tab=formatter&pane=Diagnostics&code=PAA%2FAHgAbQBsACAAdgBlAHIAcwBpAG8AbgA9ACIAMQAuADAAIgAgAGUAbgBjAG8AZABpAG4AZwA9ACIAVQBUAEYALQA4ACIAIABzAHQAYQBuAGQAYQBsAG8AbgBlAD0AIgBuAG8AIgA%2FAD4ACgA8AHMAdgBnAAoACQB3AGkAZAB0AGgAPQAiADMAMAAwACIACgAJAGgAZQBpAGcAaAB0AD0AIgAzADAAMAAiAAoACQB2AGkAZQB3AEIAbwB4AD0AIgAwACAAMAAgADMAMAAwACAAMwAwADAAIgAKAAkAdgBlAHIAcwBpAG8AbgA9ACIAMQAuADEAIgAKAAkAeABtAGwAbgBzADoAeABsAGkAbgBrAD0AIgBoAHQAdABwADoALwAvAHcAdwB3AC4AdwAzAC4AbwByAGcALwAxADkAOQA5AC8AeABsAGkAbgBrACIACgAJAHgAbQBsAG4AcwA6AHMAdgBnAD0AIgBoAHQAdABwADoALwAvAHcAdwB3AC4AdwAzAC4AbwByAGcALwAyADAAMAAwAC8AcwB2AGcAIgAKAAkAeABtAGwAbgBzAD0AIgBoAHQAdABwADoALwAvAHcAdwB3AC4AdwAzAC4AbwByAGcALwAyADAAMAAwAC8AcwB2AGcAIgAKAAkAPgAKAAkAPAB0AGkAdABsAGUAPgBCAGkAbwBtAGUAIABQAGEAcgBzAGUAcgAgAHQAZQBzAHQAIABmAG8AcgAgAEMARABBAFQAQQA8AC8AdABpAHQAbABlAD4ACgAJADwAZABlAHMAYwByAGkAcAB0AGkAbwBuAD4AQgBpAG8AbQBlACAAcABhAHIAcwBlAHIAIAAoAHYAMgAuADUALgA4ACkAIABjAGgAbwBrAGUAcwAgAHcAaABlAG4AIABlAG4AYwBvAHUAbgB0AGUAcgBpAG4AZwAgAHQAaABlACAAQwBEAEEAVABBACAAcwBlAGMAdABpAG8AbgAsACAAZgBvAHUAbgBkACAAagB1AHMAdAAgAGkAbgBzAGkAZABlACAAdABoAGUAIABmAG8AbABsAG8AdwBpAG4AZwAgAGAAcwB0AHkAbABlAGAAIABlAGwAZQBtAGUAbgB0AC4APAAvAGQAZQBzAGMAcgBpAHAAdABpAG8AbgA%2BAAoACgAJADwAcwB0AHkAbABlACAAdAB5AHAAZQA9ACIAdABlAHgAdAAvAGMAcwBzACIAPgA8ACEAWwBDAEQAQQBUAEEAWwAKAAkACQAvACoAIABDAG8AbQBtAGUAbgB0AHMAIABzAGgAbwB1AGwAZAAgAGIAZQAgAGYAaQBuAGUAIAAqAC8ACgAKAAkACQAvACoAIABJAG4AYwBsAHUAZABpAG4AZwAgAE0AdQBsAHQAaQAtAGwAaQBuAGUAIABjAG8AbQBtAGUAbgB0AHMACgAJAAkALQAtAC0ALQAtAC0ALQAtAC0ALQAtAC0ALQAtAC0ALQAtAC0ALQAtAC0ALQAtAC0ALQAtAC0ALQAtAC0ALQAtACoALwAKAAkACQBwAG8AbAB5AGcAbwBuACAAewAKAAkACQAJAHMAdAByAG8AawBlADoAIABwAHUAcgBwAGwAZQA7AAoACQAJAAkAcwB0AHIAbwBrAGUALQB3AGkAZAB0AGgAOgAgADUAcAB4AAoACQAJAH0ACgAKAAkAXQBdAD4ACgAJADwALwBzAHQAeQBsAGUAPgAKAAoACQA8AHAAbwBsAHkAZwBvAG4AIABwAG8AaQBuAHQAcwA9ACIAMgAwADcALgAxADUALAAxADEANwAgADEANQAwACwAMgAxADYAIAA5ADIALgA4ADUALAAxADEANwAiAD4ACgAJAAkAPAB0AGkAdABsAGUAPgBUAHIAaQBhAG4AZwBsAGUAPAAvAHQAaQB0AGwAZQA%2BAAoACQA8AC8AcABvAGwAeQBnAG8AbgA%2BAAoACgA8AC8AcwB2AGcAPgAKAA%3D%3D&language=html)

(Whoa…Just paste in the code and the URL updates itself!  That’s awesome! :D )

**Comment by Zearin** [comment]
> What's the other tool you're using?

The [Scholarly XML plugin for VSCodium](https://open-vsx.org/vscode/item?itemName=raffazizzi.sxml).

**Comment by Zearin** [comment]
@biome-cookie I added a link to the playground. Are you able to remove the label `S-Needs repro`?

**Comment by dyc3** [comment]
The bug is that the contents of `<style>` aren't being treated as verbatim, not that CDATA needs to be parsed inside a style element.

**Comment by Zearin** [comment]
Oh…?!  

I believe you (of course!)…  But if you remove the starting and ending CDATA items, the diagnostic errors go away.  That’s what led me to believe it was the CDATA itself.

**Comment by dyc3** [comment]
Actually, I made a mistake. The diagnostic is actually being emitted by the CSS parser, which is expected behavior. The contents of `<style>` elements is expected to be css by default. CDATA sections are not valid CSS, and therefore should emit the diagnostic.

_However_, it appears that in SVG contexts, CDATA sections are parsed even in verbatim sections like `<style>`. I am not a fan of this because of the complexity it implies.
